.github/workflows/test-build-system.yml

```diff
@@ -0,0 +1,30 @@
+# This workflow will thoroughly test the build system
+name: Complete build system test
+
+on:
+  push:
+    branches: '**'
+  pull_request:
+    branches: '**'
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install build dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install --no-install-recommends make gettext gnome-shell libglib2.0-bin
```

Comment:
Why do you need `gnome-shell` ?

Reply:
Good catch, we don't. This was ported from my extensions, where I used `gnome-extensions pack` and `gnome-extensions install` for the `package` and `install` targets. It might be worth porting to `gnome-extensions` at some point, as that might catch issues like #244 before release, but I can do that in another PR if you'd rather.

Reply:
If `gnome-extensions` provides more guarantees why not. What's bothering me is downloading `gnome-shell` + all dependencies for a simple check...

Reply:
The `--no-install-recommends` should help mitigate the number of dependencies. This workflow could be grouped with the test build & upload, to avoid setting up the same dependencies twice.

Reply:
Still there's a lot of dependencies and it's a waste of resources to download them over and over again on each run. At least we should setup some cache for apt packages (ex: https://github.com/awalsh128/cache-apt-pkgs-action)

If the behaviour of `gnome-extensions` changes from one gnome-shell release of another (not yet officially released) I doubt we will catch it as we will still use the older version. There's probably no good solution here.

Anyway I think it's a good idea to standardize the build/install with the tool provided by gnome-shell.

Reply:
Ok, I'll look into the caching the dependencies. True, the workflow probably won't catch issues due to changes in new versions, but manual testing would catch this on the target version, as long as `gnome-extensions`. is used.